pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/BasePinotFS.java

```diff
@@ -36,6 +36,11 @@ public boolean deleteBatch(List<URI> segmentUris, boolean forceDelete)
       throws IOException {
     boolean result = true;
     for (URI segmentUri : segmentUris) {
+      // Check if file exists before attempting deletion to avoid FileNotFoundException
+      if (!exists(segmentUri)) {
```

Comment:
This makes the contract for `deleteBatch()` and `delete()` different.

We need to decide the behavior for the following scenarios:
- For `delete()`
  - File exists
    - File is dir
      - Deleted
      - Not deleted
    - File is ordinary file
      - Deleted
      - Not deleted
  - File doesn't exist
- For `deleteBatch()`
  - All files deleted
  - Some files deleted
  - None file deleted

@swaminathanmanish @KKcorps @abhishekbafna Can you please also comment here?

Reply:
Going by the `idempotent` property, successful deletion and missing file, both should return `true`, otherwise `false`.

For the batch, if all the files are deleted or removed, return `true` otherwise `false`.